pkg/sqlstore/impl/user/rowstojson.go

```diff
@@ -45,6 +47,19 @@ func getRowsData(rows *sql.Rows, numColumns int) ([][]interface{}, error) {
 		if err := rows.Scan(scanArgs...); err != nil {
 			return nil, fmt.Errorf("scan row column: %s", err)
 		}
+		for i, scanArg := range scanArgs {
+			switch src := (*scanArg.(*interface{})).(type) {
+			case string:
+				trimmed := strings.TrimLeft(src, " ")
+				if (strings.HasPrefix(trimmed, "{") || strings.HasPrefix(trimmed, "[")) && json.Valid([]byte(trimmed)) {
+					scanArgs[i] = []byte(trimmed)
+				}
+			case []byte:
+				tmp := make([]byte, len(src))
+				copy(tmp, src)
+				scanArgs[i] = tmp
```

Comment:
Not clear to me this is doing anything? Like, if we didn't do this PR at all, `[]byte` values would be handled fine, so I'm not sure why we need to do this.

Reply:
This is not putting in `scanArgs[i]` the bytes returned by the driver, since the driver might reuse those bytes in the next row (override).
This is something that wasn't right before this PR.

Maybe `mattn/sqlite3` doesn't reuse the bytes for each row since that's just a constraint put at the interface level of `database/sql`, so maybe some drivers prefer to lean on the safe side.

Reply:
I also not entirely sure how many people have really used `BLOB` columns to make this potential bug be noticed.